src/Top_opt_3d/top3d.jl

```diff
@@ -0,0 +1,324 @@
+        @timeit to "print_results" print_results(loop, c, xPhys, change);
+
+        # PLOT DENSITIES
+            if displayflag
+                @timeit to "display_3D" display_3D(xPhys, displayflag);
+            end
+    end
+    xPhys;
+
+end
+
+function top3d(nelx, nely, nelz, volfrac, penal, rmin)
+
+    @timeit to "top3d" begin
+        
+        # USER-DEFINED LOOP PARAMETERS
+        @timeit to "user_defined_loop_parameters" maxloop, tolx, displayflag = user_defined_loop_parameters();
+
+        # USER-DEFINED MATERIAL PROPERTIES
+        @timeit to "user_defined_material_properties" E0, Emin, nu = user_defined_material_properties();
+
+        # USER-DEFINED LOAD DOFs
+        @timeit to "user_defined_load_dofs" loaddof = user_defined_load_dofs(nelx, nely, nelz);
+
+        # USER-DEFINED SUPPORT FIXED DOFs
+        @timeit to "user_defined_support_fixed_dofs" fixeddof = user_defined_support_fixed_dofs(nelx, nely, nelz);
+
+        # PREPARE FINITE ELEMENT ANALYSIS
+        @timeit to "prepare_finite_element_analysis" nele, F, U, freedofs, KE, edofMat, iK, jK = prepare_finite_element_analysis(nelx, nely, nelz, loaddof, fixeddof, nu);
+
+        # PREPARE FILTER
+        @timeit to "prepare_filter" H, Hs = prepare_filter(nelx, nely, nelz, nele, rmin);
+
+        # OPTIMIZE
+        @timeit to "optimize" xPhys = optimize(volfrac, nelx, nely, nelz, maxloop, tolx, Emin, penal, E0, nele, KE, iK, jK, freedofs, edofMat, H, Hs, displayflag, F, U);
+
+        # PLOT
+        @timeit to "display_3D" display_3D(xPhys);
+
+    end
+end
+
+function meshgrid(x, y, z)
+    X = zeros(size(y, 1), size(x, 1), size(z, 1))
+    Y = zeros(size(y, 1), size(x, 1), size(z, 1))
+    Z = zeros(size(y, 1), size(x, 1), size(z, 1))
+    X .= x' .* ones(size(y, 1)) .* ones(1, 1, size(z, 1));
+    Y .= ones(size(x, 1))' .* y .* ones(1, 1, size(z, 1));
+    Z .= reshape(z, 1, 1, :) .* ones(size(y, 1), size(x, 1));
+    X, Y, Z
+end
+
+# === DISPLAY 3D TOPOLOGY (ISO-VIEW) ===
+function display_3D(rho, displayflag=false)
+    nely, nelx, nelz = size(rho);
+    hx = 1; hy = 1; hz = 1;            # User-defined unit element size
+    scene = Figure();
+    flag = false;
+    faces = [ 1 2 3; 3 4 1;
+    5 6 7; 7 8 5;
+    6 2 3; 3 7 6;
+    5 1 4; 4 8 5;
+    8 7 3; 3 4 8;
+    5 6 2; 2 1 5 ]
+    for k = 1:nelz
+        z = (k-1)*hz;
+        for i = 1:nelx
+            x = (i-1)*hx;
+            for j = 1:nely
+                y = nely*hy - (j-1)*hy;
+                if (rho[j,i,k] > 0.7)  # User-defined display density threshold
+                    vert = [x y z; x y-hx z; x+hx y-hx z; x+hx y z; x y z+hx; x y-hx z+hx; x+hx y-hx z+hx; x+hx y z+hx];
+                    vert[:, 2:3] = vert[:, 3:-1:2]; vert[:, 2, :] = -vert[:, 2, :];
+                    if flag == false
+                        scene = mesh(vert, faces, color = RGB(0.2+0.8*(1-rho[j,i,k]), 0.2+0.8*(1-rho[j,i,k]), 0.2+0.8*(1-rho[j,i,k])), shading = NoShading; axis=(; show_axis = false))
```

Comment:
this doesn't work on my machine, there are 2 issues:
- you can't have kwargs before the `;`
- I think `shading` should be boolean `true/false`